crates/assessor/src/lib.rs

```diff
@@ -77,15 +83,28 @@ impl Fulfillment {
     }
     /// Evaluates the requirements of the request.
     pub fn evaluate_requirements(&self) -> Result<(), Error> {
-        if !self.request.requirements.predicate.eval(&self.journal) {
-            return Err(Error::PredicateEvaluationError);
+        let requirements = &self.request.requirements;
+
+        if requirements.predicate.predicateType == PredicateType::ClaimDigestMatch
+            && requirements.callback.callbackType != CallbackType::None
+        {
+            return Err(Error::RequirementsEvaluationError);
+        }
```

Comment:
I would drop this constraint (partly because my suggestion is to drop `CallbackType`) and instead fail when trying to construct the `AssessorCallback` is the `FullfillmentClaimData` does not contain the image ID and journal. It may need somewhat odd that this would enable a `ClaimDigestMatch` request to actually have a callback, but this is actually fine so long as we check that `ReceiptClaim::ok(image_id, journal).digest()` does in fact meet the requirements, which we do.